### ISSUE TITLE
Problem parsing Macros ending in a semicolon (;)

### ISSUE THREAD

**Issue by blackfisch** [SkaceKamen/vscode-sqflint#115]
SQFLint throws a random Error `Expected ";" but "\n" found`, if a macro ends with a semicolon.

Example:

```cpp
#define A(NAME) class NAME {};

A(test1)
A(test2)
```
above throws the mentioned error, rewriting it like this does not:

```cpp
#define A(NAME) class NAME {}

A(test1);
A(test2);
```

It's an easy workaround but probably quite annoying and honestly I just found this by accident. Maybe this can be fixed. Something along the lines of "if the last character of the macro is a semicolon, skip semicolon check."

**Comment by blackfisch** [comment]
> Not related to the issue but worth mentioning that semicolons at end of the macros are considered a bad practice.

good point, many frameworks etc use it tho.
After rewriting all macros everything works as intended. I can see it generating a warning about it being bad practise, it should however not stop the linter from working
